test/unit/app/controllers/incoming-transactions-test.js

```diff
@@ -139,28 +182,443 @@ describe('IncomingTransactionsController', function () {
+      )
+    })
+
+    it('should update last block fetched when started and not on supported network', async function () {
+      const incomingTransactionsController = new IncomingTransactionsController({
+        blockTracker: getMockBlockTracker(),
+        networkController: getMockNetworkController(),
+        preferencesController: getMockPreferencesController(),
+        initState: getNonEmptyInitState(),
+      })
+      const updateStateStub = sinon.stub(incomingTransactionsController.store, 'updateState')
+      const updateStateCalled = waitUntilCalled(updateStateStub, incomingTransactionsController.store)
+
+      incomingTransactionsController.start()
+
+      await updateStateCalled
+
+      const state = incomingTransactionsController.store.getState()
+      assert.deepStrictEqual(
+        state,
+        {
+          incomingTransactions: {
+            ...getNonEmptyInitState().incomingTransactions,
+          },
+          incomingTxLastFetchedBlocksByNetwork: {
+            ...getNonEmptyInitState().incomingTxLastFetchedBlocksByNetwork,
+            [FAKE_NETWORK]: 11,
+          },
+        },
+        'Should update last block fetched',
+      )
+    })
+
+    it('should not update upon latest block when started and incoming transactions disabled', async function () {
+      const incomingTransactionsController = new IncomingTransactionsController({
+        blockTracker: getMockBlockTracker(),
+        networkController: getMockNetworkController(),
+        preferencesController: getMockPreferencesController({ showIncomingTransactions: false }),
+        initState: getNonEmptyInitState(),
+      })
+      // reply with a valid request for any supported network, so that this test has every opportunity to fail
+      for (const network of [GOERLI, KOVAN, MAINNET, RINKEBY, ROPSTEN]) {
+        nock(`https://api${network === MAINNET ? '' : `-${network.toLowerCase()}`}.etherscan.io`)
+          .get(/api.+/u)
+          .reply(
+            200,
+            JSON.stringify({
+              status: '1',
+              result: [getFakeEtherscanTransaction()],
+            }),
+          )
+      }
+      const updateStateStub = sinon.stub(incomingTransactionsController.store, 'updateState')
+      const updateStateCalled = waitUntilCalled(updateStateStub, incomingTransactionsController.store)
+      const putStateStub = sinon.stub(incomingTransactionsController.store, 'putState')
+      const putStateCalled = waitUntilCalled(putStateStub, incomingTransactionsController.store)
+
+      incomingTransactionsController.start()
+
+      try {
+        await Promise.race([
+          updateStateCalled,
+          putStateCalled,
+          new Promise((_, reject) => {
+            setTimeout(() => reject(new Error('TIMEOUT')), 1000)
+          }),
+        ])
+        assert.fail('Update state should not have been called')
+      } catch (error) {
+        assert(error.message === 'TIMEOUT', 'setState was not called')
```

Comment:
nit:
```suggestion
        assert(error.message === 'TIMEOUT', 'setState should not have been called')
```

Reply:
Thanks, good catch. That old description was wrong. I think I'll change this to something like "timeout exception should be thrown" though.

If this assertion fails, it's still unclear whether `setState` was called or not, because the exception thrown may have been before, after, or within `setState`.

Reply:
Addressed here: https://github.com/MetaMask/metamask-extension/pull/9755/commits/28a3d43098e8ae22a8e6176f7ba17f9bda0887e3